pyensign/ensign.py

```diff
@@ -474,3 +489,203 @@ def _resolve_topic(self, topic):
             raise UnknownTopicError(
                 f"unknown topic '{topic}', please provide the name or ID of a topic in your project"
             )
+
+
+##########################################################################
+## Decorators
+##########################################################################
+
+# Global Ensign client. This is not thread-safe but it's safe to use concurrently in
+# asyncio coroutines.
+_client = None
+
+
+def authenticate(*auth_args, **auth_kwargs):
```

Comment:
So this is like an alternative way to use PyEnsign. Instead of creating the client directly, you can decorate functions with `@publish`. This does create an interesting question about how to authenticate the user, so my current solution was to add an `@authenticate` decorator as well. This decorator can be put on any async function, so that any function under it in the call stack uses the same client for authentication.